pkg/detect/detect.go

```diff
@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+package detect
+
+import (
+	"bufio"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/redhat-partner-solutions/vse-sync-collection-tools/pkg/clients"
+	"github.com/redhat-partner-solutions/vse-sync-collection-tools/pkg/collectors/contexts"
+	"github.com/redhat-partner-solutions/vse-sync-collection-tools/pkg/utils"
+)
+
+type DetectedInterface struct {
+	Name            string `json:"name"`
+	PTPClockDevPath string `json:"ptp_dev"` //nolint:tagliatelle // script assumes
```

Comment:
Not sure what the variable is for

Reply:
It is filled by the detection tool from the ptp clock number from `ethtool -T <interface>` it is used by the `e2e.sh` in the tests repo to define the ptp clock in `ts2phc` logs